src/language-js/comments.js

```
@@ -578,7 +578,7 @@ function handleCommentAfterArrowParams(text, enclosingNode, comment, options) {
     comment,
     options.locEnd
   );
-  if (text.slice(index, index + 2) === "=>") {
+  if (index !== false && text.slice(index, index + 2) === "=>") {
```

```suggestion

  // @ts-ignore
  if (text.slice(index, index + 2) === "=>") {
```

to back out the extra condition, ignore the error, and add a blank line to improve readability?

---

No, extra check is good.

---

Keeping these extra checks then, thanks.